### PR TITLE
Add item groups (aka creative tabs)

### PR DESCRIPTION
**Pull request by ghost**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1866) | **SpongeCommon** | [SpongeForge](https://github.com/SpongePowered/SpongeForge/pull/2327)

* Implemented SpongeAPI changes
* Added `ItemGroupRegistryModule` and `ItemGroupTest`

### CODE REVIEW

File: testplugins/src/main/java/org/spongepowered/test/ItemGroupTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.spongepowered.test;
```
Comment (kashike) [inline testplugins/src/main/java/org/spongepowered/test/ItemGroupTest.java:25]: Missing license header.

**Review by kashike — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/common/mixin/core/creativetab/MixinCreativeTabs.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public String getName() {
+        return getId();
+    }
+
+    @Override
+    public Translation getTranslation() {
+        return new SpongeTranslation("itemGroup." + this.tabLabel);
```
Comment (ImMorpheus) [inline src/main/java/org/spongepowered/common/mixin/core/creativetab/MixinCreativeTabs.java]: Translations should be lazy loaded

File: src/main/java/org/spongepowered/common/mixin/core/creativetab/MixinCreativeTabs.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Shadow @Final private String tabLabel;
+
+    @Override
+    public String getId() {
+        return this.tabLabel;
+    }
+
+    @Override
+    public String getName() {
+        return getId();
```
Comment (ImMorpheus) [inline src/main/java/org/spongepowered/common/mixin/core/creativetab/MixinCreativeTabs.java]: https://github.com/SpongePowered/SpongeCommon/blob/df3266d7b2923018f111df3a4db442e6a4882841/src/main/java/org/spongepowered/common/SpongeCatalogType.java#L33

Once https://github.com/SpongePowered/SpongeCommon/pull/1792 is merged, you'll see the warning during the tests.

**Comment by gbui** [comment]
TODO: open pull requests for SpongeVanilla and SpongeForge implementations

**Comment by ghost** [comment]
~wip

**Comment by ghost** [comment]
~qa

**Comment by gabizou** [comment]
Since bleeding (API 8) will be targeting Minecraft 1.13, I'm going to keep the implementation not merged until we have a 1.13 workspace ready. I'm not wanting to squash the API, I just want the API to be implementable when we are getting 1.13 to work, instead of having to rip the API apart again. I haven't personally checked if there's much changes regardless.

**Pull request closed without merging**